.github/workflows/build_and_test.yml

```diff
@@ -1644,26 +1634,12 @@ jobs:
         . setenv.sh
         cd tests/cmake_integration
         perl run_ci_tests.pl --skip-cxx11
```

Comment:
This and all the other jobs that are using the non default `g++` need the compiler explicitly set because of the issue I mentioned [here](https://github.com/objectcomputing/OpenDDS/pull/2526#discussion_r605277436).